frontend/src/Editor/Viewer/MobileHeader.jsx

```diff
@@ -30,6 +31,13 @@ const MobileHeader = ({
     }),
     shallow
   );
+  const { appMode } = useEditorStore(
+    (state) => ({
+      appMode: state.appMode,
+    }),
+    shallow
+  );
+  const showDarkModeToggle = useMemo(() => appMode === 'auto', [appMode]);
```

Comment:
`const { appMode, showDarkModeToggle } = useEditorStore(
    (state) => ({
      appMode: state.appMode,
      showDarkModeToggle: state.appMode === 'auto'
    }),
    shallow
  );`

Check for the feasibility. This way we can avoid `useMemo`